### DISCUSSION TITLE
update to v0.49.29 cause "failed to resolve component" issue

### DISCUSSION THREAD

**Discussion by zillionare** [slidevjs/slidev#1863] [Q&A]
Not sure if it's a bug, my project works under v0.49.16. After upgrade to v0.49.29, the following error raised:

```
[Vue warn]: Failed to resolve component: Thebe
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

Thebe is a custom element made myself, to connect slidev with jupyter server, to run python code on jupyter server and fetch result back.

**Comment by zillionare** [comment]
I have rollback to 0.49.16 by:

1. rm -rf node_modules
2. rm package-lock.json
3. modify @slidev/cli version to exactly 0.49.16 in package.json
4. run npm install

the error is gone.

does anyone know are there any broken changes during 0.49.16 and 0.49.29?

**Reply by zillionare** [reply]
after rollbacking to 0.49.16, another issue raise, the hmr issue (#1821 ).

not sure what happened here, it all works a week ago. Yesterday I tried to enable monaco editor, then find lz-string issue (#1861 ), then trying to upgrade slidev, then all components made by myself cannot work all of sudden.

Is this caused by dependency slidev uses, like vite?

**Reply by kermanx** [reply]
Could you help identify the exact version when the component resolution problem started to happen?

**Reply by zillionare** [reply]
I am currently checking, but I am not an expert and sometimes I can be deceived by false appearances. For example, I might think everything is working fine, but it could actually be due to caching.

I can confirm that my project was still working on the night of September 9 (China time), because I gave a demonstration at that time and used several custom components, such as the Iframe and Carousel components that I wrote. So, at that point, there were no issues. The version I can trace back to is 0.49.16, which may have had fewer HMR issues. However, when I reverted to 0.49.16 today, the HMR problem became completely reproducible.

I am currently trying to find a working version, and I will update this once I have results. To avoid unexpected errors, my approach is to first delete the .vite directory, then delete the node_modules directory, modify the @slidev/cli version, and reinstall. Will this process ensure that there are no unexpected issues? Additionally, Vite has been updated recently. If @slidev/cli does not lock the Vite version, reinstalling @slidev/cli could result in updating Vite, even if the @slidev/cli version is still locked to an older version. This makes the search more difficult.

**Reply by kermanx** [reply]
I think the simplest way is to use https://sli.dev/new. You can firstly terminate the automat npm install command, modify package.json to use the version you want, and remove package-lock.json(if there is one), then run npm install and dev.

**Comment by zillionare** [comment]
confirmed that 0.49.16 is the latest workable version regarding to custom components resolving.

To speed the test, the final steps are:

1. modify version of slidev/cli to chosen version in package.json
2. delete package-lock.json
3. rm -rf .vite
4. npm i

** so I didn't remove node_modules **, due to my network speed is too slow.

Here is the result.

I pinned vite version as 5.3, vue as 3.5.2, founded in my another environment.

| version | component resolve                                          | hrm  |
| ------- | ---------------------------------------------------------- | ---- |
| 0.49.20 | fail                                                       | work |
| 0.49.19 | fail                                                       | work |
| 0.49.18 | fail                                                       | work |
| 0.49.17 | No matching version found for @shikijs/transformers@1.17.0 ,test failed     |
| 0.49.16 | work                                                       | fail |
| 0.49.12 | work                                                       | fail |
| 0.49.10 | work                                                       | fail |

Hope this could help.

My workaround would be (if this can help others):

1. update to 0.49.29, write most slide in this version, so I can leverage HMR
2. switch back to 0.49.16, for presentation

**Comment by zillionare** [comment]
have filed an issue as #1864 , and attached a sample for reproduce. close this.
